### PR TITLE
[Release] Update to 1.5 (#315)

### LINKED ISSUE #314: upgrade to support dbt-core v1.5.0
## Background

The latest version of dbt Core,`dbt-core==1.5.0rc1`, was published on April 13, 2023 ([PyPI](https://pypi.org/project/dbt-core/1.5.0rc1/) | [Github](https://github.com/dbt-labs/dbt-core/releases/tag/v1.5.0rc1)).

## How to upgrade

https://github.com/dbt-labs/dbt-core/discussions/7213 is an open discussion with more detailed information. If you have questions, please put them there!

The above linked guide has more information, but below is a high-level checklist of work that would enable a successful 1.5.0 release of your adapter. 

- [ ] Add support Python 3.11 (if you haven't already)
- [ ] Add support for relevant tests (there's a lot of new ones!)
- [ ] Add support model contracts
- [ ] ~~Add support for materialized views~~ (this likely will be bumped to 1.6.0)

## the next minor release: `1.6.0`

FYI, `dbt-core==1.6.0` is expected to be released at the end of July, with a release cut at least two weeks prior.

More information to come as to what to expect from that release.

### PR DESCRIPTION
**Pull request by andrefurlan-db**
Resolves #314

### Description

Update to dbt 1.5.0 with release `1.5.0rc1`.

* We fixed all tests
* Added nice error message for model contracts while we are working on supporting it

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -22,6 +22,6 @@ pytz
 tox>=3.2.0
 types-requests
 
-dbt-spark==1.4.*
-dbt-core==1.4.*
-dbt-tests-adapter==1.4.*
\ No newline at end of file
+dbt-spark==1.5.0b4
+# dbt-core==1.4.*
+dbt-tests-adapter==1.5.0rc1
```
Comment (dataders) [inline dev-requirements.txt:27]: can you try this? this basically installs based on the `setup.py` in dbt-core's `tests/adapter/` dir but from the `1.5.latest` branch.

my hunch is that something's goofy w/ `dbt-tests-adapter==1.5.0rc1` PyPI package...


```suggestion
git+https://github.com/dbt-labs/dbt-core.git@1.5.latest#egg=dbt-tests-adapter&subdirectory=tests/adapter
```

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 databricks-sql-connector>=2.5.0
-dbt-spark==1.4.*
+
+dbt-spark==1.5.0b4
```
Comment (dataders) [inline requirements.txt:3]: also this
```suggestion
git+https://github.com/dbt-labs/dbt-spark.git@1.5.latest#egg=dbt-spark
```

File: dev-requirements.txt (hunk 1 above)
Comment (dataders) [inline dev-requirements.txt:27]: can you try this? this basically installs based on the `setup.py` in dbt-core's `tests/adapter/` dir but from the `1.5.latest` branch.

my hunch is that something's goofy w/ `dbt-tests-adapter==1.5.0rc1` PyPI package...


```suggestion
git+https://github.com/dbt-labs/dbt-spark.git@1.5.latest#egg=dbt-spark
git+https://github.com/dbt-labs/dbt-core.git@1.5.latest#egg=dbt-core&subdirectory=core
git+https://github.com/dbt-labs/dbt-core.git@1.5.latest#egg=dbt-tests-adapter&subdirectory=tests/adapter
```

**Review by rcypher-databricks — APPROVED** [review]

**Comment by andrefurlan-db** [comment]
I'll merge this with Ray's approval

**Pull request merged**